web/src/engine/keyboard/src/kmxPlus/convertLayerList.ts

```diff
@@ -0,0 +1,122 @@
+import { Layer, LayerList } from './layr.js';
+import { TouchLayout } from '@keymanapp/common-types';
+import TouchLayoutLayer = TouchLayout.TouchLayoutLayer;
+import TouchLayoutRow = TouchLayout.TouchLayoutRow;
+import TouchLayoutKey = TouchLayout.TouchLayoutKey;
+import TouchLayoutSubKey = TouchLayout.TouchLayoutSubKey;
+import TouchLayoutFlick = TouchLayout.TouchLayoutFlick;
+
+import Codes from '../codes.js';
+import { KeySpec, Keys } from './keys.js';
+import { ButtonClasses } from '../keyboards/defaultLayouts.js';
+
+const keyNames = Object.keys(Codes.keyCodes);
+
+export function convertLayerList(layerList: LayerList, keys: Keys) {
```

Comment:
This function converts one layer list (one pre-parsed entry of the `layr.lists` subtable) into the OSK-friendly `TouchLayout` format.